.evergreen/config.yml

```
@@ -2411,6 +2527,45 @@ task_groups:
             $DRIVERS_TOOLS/.evergreen/csfle/gcpkms/delete-instance.sh
     tasks:
       - testgcpkms-task
+  - name: testazurekms_task_group
+    setup_group_can_fail_task: true
+    setup_group_timeout_secs: 1800 # 30 minutes
+    setup_group:
+      - func: fetch-source
+      - func: prepare-resources
+      - func: fix-absolute-paths
+      - func: make-files-executable
+      - command: shell.exec
+        params:
+          shell: "bash"
+          script: |
+            ${PREPARE_SHELL}
+            export AZUREKMS_VMNAME_PREFIX="GODRIVER"
+            export AZUREKMS_CLIENTID=${AZUREKMS_CLIENTID}
+            export AZUREKMS_TENANTID=${AZUREKMS_TENANTID}
+            export AZUREKMS_SECRET=${AZUREKMS_SECRET}
+            export AZUREKMS_DRIVERS_TOOLS=$DRIVERS_TOOLS
+            export AZUREKMS_RESOURCEGROUP=${AZUREKMS_RESOURCEGROUP}
+            echo '${testazurekms_publickey}' > /tmp/testazurekms.pubkey
+            export AZUREKMS_PUBLICKEYPATH=/tmp/testazurekms.pubkey
+            echo '${testazurekms_privatekey}' > /tmp/testazurekms.prikey
+            sudo chmod 600 /tmp/testazurekms.prikey
+            export AZUREKMS_PRIVATEKEYPATH=/tmp/testazurekms.prikey
+            export AZUREKMS_SCOPE=${AZUREKMS_SCOPE}
+            $DRIVERS_TOOLS/.evergreen/csfle/azurekms/create-and-setup-vm.sh
+      - command: expansions.update
+        params:
+          file: testazurekms-expansions.yml
+    teardown_group:
```

```suggestion
    teardown_group:
      # setup_group may have failed before updating expansions. Try to update again to clean up resources on setup failure.
      - command: expansions.update
        params:
          ignore_missing_file: true
          file: testazurekms-expansions.yml
```

If the `setup_group` fails, the `expansions.update` may not have been run. That may result in resources not being destroyed.